cpp/include/rapidsmpf/all_gatherer/all_gatherer.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2024-2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <rapidsmpf/buffer/packed_data.hpp>
+#include <rapidsmpf/buffer/resource.hpp>
+#include <rapidsmpf/communicator/communicator.hpp>
+#include <rapidsmpf/progress_thread.hpp>
+#include <rapidsmpf/shuffler/shuffler.hpp>
+#include <rapidsmpf/statistics.hpp>
+
+namespace rapidsmpf::experimental::all_gatherer {
```

Comment:
I don't think er need `experimental`, everything is experimental at this point :)